public/src/card/sectioned-card-list.html

```diff
@@ -0,0 +1,211 @@
+<link rel="import" href="../../bower_components/polymer/lib/elements/dom-repeat.html">
+<link rel="import" href="card-list.html">
+
+<dom-module id="sectioned-card-list">
+  <template>
+    <style is="custom-style" include="app-styles">
+      :host {
+        display: block;
+        box-sizing: border-box;
+      }
+
+      card-list {
+        max-width: 1264px;
+        margin: 10px auto;
+        box-sizing: border-box;
+      }
+
+      .promotedPanel {
+        background: #fafafa;
+        padding: 2px 0 10px;
+        border-top: 1px solid #f0f0f0;
+        border-bottom: 1px solid #f0f0f0;
+      }
+
+      #more {
+        box-sizing: border-box;
+        padding: 30px 0 20px;
+      }
+
+      #moreButton {
+        text-align: center;
+        width: 85px;
+        height: 85px;
+        cursor: pointer;
+        margin: 0 auto;
+        border-radius: 50%;
+        font-size: 16px;
+        text-transform: uppercase;
+        font-weight: bold;
+        letter-spacing: 0.05em;
+        color: var(--highlight-green);
+        background: transparent;
+        border: 3px solid;
+      }
+
+      @media (max-width: 1264px) {
+        card-list {
+          max-width: 948px;
+        }
+      }
+
+      @media (max-width: 948px) {
+        card-list {
+          max-width: 632px;
+        }
+      }
+
+      @media (max-width: 632px) {
+        card-list {
+          max-width: 320px;
+        }
+      }
+    </style>
+    <dom-repeat items="[[sections]]">
+      <template>
+        <div class$="{{getPanelClass(item)}}">
+          <card-list label="[[item.label]]" items="[[item.cards]]" more="[[item.more]]"></card-list>
+        </div>
+      </template>
+    </dom-repeat>
+    <div id="more" class="horizontal layout center hidden">
+      <div id="moreButton" class="horizontal layout center" on-click="_onMore">
+        <div style="width: 100%; box-sizing: border-box;">
+          <div>More</div>
+        </div>
+      </div>
+    </div>
+  </template>
+  <script>
+    class SectionedCardList extends Polymer.Element {
+      static get is() { return 'sectioned-card-list'; }
+      static get properties() {
+        return {
+          items: {
+            type: Array,
+            observer: '_onItems'
+          },
+          sections: Array,
+          more: {
+            type: Boolean,
+            observer: '_refreshMore'
+          },
+          label: String
+        };
+      }
+
+      constructor() {
+        super();
+        this._resizeListener = _debounce(() => {
```

Comment:
Curious.  Shouldn't this really be in attached/detached rather than constructor and disconnected?

Reply:
This is like defining a function definition or property. Attached/detached can be called multiple times. In this case it doesn't matter. Generally shouldn't be doing any DOM stuff in the constructor. This is just defining a class member.